src/ovp_pipeline/commands/_ui_renderers.py

```
@@ -500,6 +488,82 @@ def _object_href(object_id: str, path: str = "", requested_pack: str = "") -> st
+def _anchor_title_for_note(relative_path: str, markdown: str) -> str:
+    """Return the friendly title for a note anchor.
+
+    Prefers the H1 line from the markdown body, then the YAML
+    ``title:`` field *from the frontmatter block only* (CodeRabbit
+    M — searching the whole body would pick up an example code
+    fence's ``title:`` line), then the path basename.
+    """
+    body = markdown
+    try:
+        frontmatter, body = _parse_frontmatter(markdown)
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)